apis/slo/v1alpha1/nodeslo_types.go

```diff
@@ -23,6 +23,109 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// MemoryQOS enables memory qos features based on Anolis OS.
+type MemoryQOS struct {
+	// TODO: check anolis os version
+	// memcg qos
+	// If enabled, memcg qos will be set by the agent, where some fields are implicitly calculated from pod spec.
+	// 1. `memory.min` := spec.requests.memory * minLimitFactor / 100 (use 0 if requests.memory is not set)
+	// 2. `memory.low` := spec.requests.memory * lowLimitFactor / 100 (use 0 if requests.memory is not set)
+	// 3. `memory.limit_in_bytes` := spec.limits.memory (set $node.allocatable.memory if limits.memory is not set)
+	// 4. `memory.high` := memory.limit_in_bytes * throttlingFactor / 100 (use "max" if memory.high <= memory.min)
+	// MinLimitPercent specifies the minLimitFactor percentage to calculate `memory.min`, which protects memory
+	// from global reclamation when memory usage does not exceed the min limit.
+	// Cluster default: 100; Close: 0.
+	MinLimitPercent *int64 `json:"minLimitPercent,omitempty"`
```

Comment:
add default validation

Reply:
to simplify internal protocols, we'd like to hide the default in a nodeSLO CR

Reply:
not necessary now since nodeslo is an internal protocal